### PR TITLE
fix potential buffer overflow.

### PR DESCRIPTION
**Pull request by ihsinme**
elsewhere, record size is tightly controlled.
in this place it seemed to me that no.

I suggest a quick and easy way to fix it.

### CODE REVIEW

File: src/libopensc/reader-pcsc.c [hunk 1]
```diff
@@ -585,7 +585,7 @@ static void initialize_uid(sc_reader_t *reader)
 		if (SC_SUCCESS == pcsc_transmit(reader, &apdu)
 				&& apdu.sw1 == 0x90 && apdu.sw2 == 0x00) {
 			reader->uid.len = apdu.resplen;
-			memcpy(reader->uid.value, apdu.resp, reader->uid.len);
+			memcpy(reader->uid.value, apdu.resp, MIN(sizeof(reader->uid.value), reader->uid.len));
```
Comment (frankmorgner) [inline src/libopensc/reader-pcsc.c:588]: uid.len is 0 at this point, so your fix won't work. Also, there are some corner cases that aren't catched

**Review by frankmorgner — CHANGES_REQUESTED** [review]
Thanks for noticing this issue. However, I've proposed a different approach in https://github.com/OpenSC/OpenSC/pull/2232/ , which should hopefully catch all corner cases...

File: src/libopensc/reader-pcsc.c (hunk 1 above)
Comment (Jakuje) [reply src/libopensc/reader-pcsc.c:588]: This is missing the adjustment of the `reader->uid.len` value if we get longer response. I like the #2232 better as it handles the issue outside of the block.

**Comment by ihsinme** [comment]
your fix looks like other such places in your code.
this is probably the best solution at this stage.
in any case, the main thing is to get a more secure version.

**Comment by frankmorgner** [comment]
thank you

**Comment by ihsinme** [comment]
Patch reward submission 408: OpenSC/OpenSC buffer overflow for OpenSC/OpenSC

**Pull request closed without merging**